### ISSUE TITLE
device cpp_standalone fails for bipolar_cell example

### ISSUE THREAD

**Issue by moritzaugustin** [brian-team/brian2#558]
When running bipolar_cell using the device cpp_standalone I get the following error (current master branch=commit fd3afff2be7b506d8aa76e6fd2259f888fa40ba0): 
```  
File "bipolar_cell_cpp_standalone.py", line 39, in <module>
    neuron.I[morpho.L[50*um]] = 0.2*nA  # injecting in the left dendrite
  File "/home/augustin/Projects/Brian2/github/brian-team_brian2/brian2/core/variables.py", line 886, in __setitem__
    self.set_item(item, value, level=1)
  File "/home/augustin/Projects/Brian2/github/brian-team_brian2/brian2/core/variables.py", line 883, in set_item
    check_units=check_units)
  File "/home/augustin/Projects/Brian2/github/brian-team_brian2/brian2/core/base.py", line 251, in device_override_decorated_function
    return getattr(curdev, name)(*args, **kwds)
  File "/home/augustin/Projects/Brian2/github/brian-team_brian2/brian2/devices/cpp_standalone/device.py", line 347, in variableview_set_with_index_array
    elif len(value) != len(indices):
TypeError: len() of unsized object
```
The potentially most relevant variables in device.py have the following values directly before the TypeError is raised:
```
arrayname = _array_spatialneuron_I
item = <brian2.spatialneuron.morphology.Cylinder object at 0x7f44c6359710>
value = 200. pA
indices = 125
```

**Comment by mstimberg** [comment]
Ah, this is annoying. I think the issue is fixed in the `morphology_refactoring` branch, but that branch is not quite ready yet. As a workaround, it should work if you replace the line that throws the error by:
```Python
neuron.I[morpho.L.indices[50*um]] = 0.2*nA  # injecting in the left dendrite
```

**Comment by mstimberg** [comment]
This has been fixed a while ago (8194a202), closing this issue.
